src/cast_core/src/CrudActor.cpp

```diff
@@ -1044,15 +1048,24 @@ std::string getDbName(const PhaseContext& phaseContext) {
 namespace genny::actor {
 
 struct CrudActor::PhaseConfig {
-    mongocxx::collection collection;
     std::vector<std::unique_ptr<BaseOperation>> operations;
     metrics::Operation metrics;
+    std::optional<std::string> collectionName;
+    std::optional<int64_t> numCollections;
```

Comment:
consider `struct CollectionParams { string name; int num }` and then `std::optional<CollectionParams>`. Same exact binary layout but imho slightly clearer that one optional is present iff the other is.

Bonus you could have a `CollectionParams(const Node&)` ctor and do the throwing there (cheap thrills).

(You could potentially even make the `generateCollectionName` a member-function of CollectionParams. but potentially diminishing returns the further this goes.)

Reply:
yeah, that sounds cleaner.